numpy/core/tests/test_longdouble.py

```diff
@@ -30,8 +30,11 @@ def test_scalar_extraction():
 
 
 def test_repr_roundtrip():
-    o = 1 + np.finfo(np.longdouble).eps
-    assert_equal(np.longdouble(repr(o)), o,
+    o = 1 + LD_INFO.eps
+    # We will only see eps in repr if within printing precision.
+    # Printing precision defined as 17 in scalartypes.c.src
+    expected = 1.0 if LD_INFO.precision > 17 else o
+    assert_equal(np.longdouble(repr(o)), expected,
                  "repr was %s" % repr(o))
```

Comment:
Yes, you're right, I was just looking at that code.  Can correct the comment and constant, but the question is, do we really want - for example - 31 digits of precision for np.longdouble (that's what double double gives, on PPC64).  If so, we'd have to adapt the scalartypes.c.src code to detect the different types of longdouble.